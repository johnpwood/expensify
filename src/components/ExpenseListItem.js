import React from 'react'
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom'

const ExpenseListItem = ({ id, description = '', amount = -1, createdAt=0 } ) => (
    <div>
        <Link to={`/edit/${id}`}><h3>{description}:</h3></Link>
        <p>Created at: {createdAt}</p>
        <p>amount: {amount}</p>
    </div>
)
ExpenseListItem.propTypes = {
    dispatch: PropTypes.func,
    id: PropTypes.string,
    description: PropTypes.string,
    amount: PropTypes.number,
    createdAt: PropTypes.number
}

export default ExpenseListItem;
