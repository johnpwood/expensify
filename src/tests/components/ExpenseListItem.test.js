import React from 'react'
import { shallow } from 'enzyme'

import ExpenseListItem from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses'

test('should render one expense passed through props', () => { 
    const wrapper = shallow(<ExpenseListItem { ...expenses[1] } />)
    expect(wrapper).toMatchSnapshot()
    const wrapper2 = shallow(<ExpenseListItem { ...expenses[2] } />)
    expect(wrapper2).toMatchSnapshot()
})
