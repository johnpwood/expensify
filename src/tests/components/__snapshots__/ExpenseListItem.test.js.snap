// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render one expense passed through props 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/bbc123"
  >
    <h3>
      Rent
      :
    </h3>
  </Link>
  <p>
    Created at: 
    -345600000
  </p>
  <p>
    amount: 
    10095
  </p>
</div>
`;

exports[`should render one expense passed through props 2`] = `
<div>
  <Link
    replace={false}
    to="/edit/cbc123"
  >
    <h3>
      Credit Card
      :
    </h3>
  </Link>
  <p>
    Created at: 
    345600000
  </p>
  <p>
    amount: 
    1950
  </p>
</div>
`;
